sdk/cosmos/azure-spring-data-cosmos/src/samples/java/com/azure/cosmos/multidatasource/DatabaseConfiguration.java

```diff
@@ -96,16 +89,17 @@ public ReactiveCosmosTemplate primaryReactiveCosmosTemplate(CosmosAsyncClient co
 
     // -------------------------Second Cosmos Client for Secondary Cosmos Account---------------------------
     @Bean("secondaryCosmosAsyncClient")
-    public CosmosAsyncClient getCosmosAsyncClient(CosmosClientConfig secondaryCosmosClientConfig) {
-        return CosmosFactory.createCosmosAsyncClient(secondaryCosmosClientConfig);
+    public CosmosAsyncClient getCosmosAsyncClient(CosmosClientBuilder secondaryCosmosClientBuilder) {
+        return CosmosFactory.createCosmosAsyncClient(secondaryCosmosClientBuilder);
     }
 
     @Bean("secondaryCosmosConfig")
     public CosmosConfig getCosmosConfig() {
         return CosmosConfig.builder()
-            .enableQueryMetrics(true)
-            .responseDiagnosticsProcessor(new ResponseDiagnosticsProcessorImplementation())
-            .build();
+                           .enableQueryMetrics(true)
+                           .responseDiagnosticsProcessor(new ResponseDiagnosticsProcessorImplementation())
+                           .database(secondaryProperties.getDatabase())
+                           .build();
```

Comment:
Customers can still use the same `cosmosConfig` instance. 
Since they will anyway pass the `databaseName` to create `CosmosTemplate` - which then picks up the database from `CosmosFactory` here : 
From `CosmosTemplate` constructor - 
`this.databaseName = cosmosFactory.getDatabaseName();`

Reply:
it's ok, but I think it's a bit confusing, actually, database2 can use the same CosmosConfig instance which created with database1. so the CosmosConfig is not really directly related to the database name.
@saragluna do you have any other ideas?

Reply:
Yes, that is true. `CosmosConfig` is not directly related to the database name. 
In fact, I was also wondering, if we can completely get rid of `database` from `CosmosConfig` and customers can directly override the method `getDatabaseName()` from `CosmosConfigurationSupport` class. 
That's what spring-data-mongodb does. They don't have any config tied to the database. 
They only have `getDatabaseName()` in `MongoConfigurationSupport` class. 

However, it will not hurt to keep it in `CosmosConfig`, because of easiness of use. 
Having `database` in `CosmosConfig` provides the user ability to use the database from the `CosmosConfig` bean. 

What are your thoughts on this @zhoufenqin @saragluna

Reply:
@kushagraThapar in my opinion, since `CosmosConfig` is not directly related to the database name so I vote for getting rid of it. Do you know in which case will the user try to get the database name?

Reply:
@saragluna - in case of custom query execution. 
take a look at this example - https://github.com/kushagraThapar/azure-sdk-for-java/blob/update_cosmos_config_spring_data_cosmos/sdk/cosmos/azure-spring-data-cosmos/src/test/java/com/azure/spring/data/cosmos/repository/integration/PageableAddressRepositoryIT.java#L165

So in that case, I think it will be much easier to Autowire the `CosmosConfig` bean. 

In the above specific case, do you know if customer can autowire `TestRepositoryConfig` bean somehow and get it from there ?

Reply:
@saragluna - I did some more testing, and I found out that in above specific case, customer can get it from `CosmosFactory`. So I am good to remove `database` from `CosmosConfig`